src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2790,16 +2819,14 @@ private ImmutableArray<Diagnostic> GetDiagnosticsForMethodBodiesInTree(SyntaxTre
                 hasDeclarationErrors: false,
                 emitMethodBodies: false,
                 diagnostics: bindingDiagnostics,
-                filterOpt: s => IsDefinedOrImplementedInSourceTree(s, tree, span),
+                filterOpt: filterTree is object ? (Predicate<Symbol>?)(s => IsDefinedOrImplementedInSourceTree(s, filterTree!, filterSpan)) : (Predicate<Symbol>?)null,
```

Comment:
>(Predicate<Symbol>?) [](start = 142, length = 20)

nit: do we need the second cast?

Reply:
Yes, this works around the bug I'm fixing in main.

---
In reply to: [596330844](https://github.com/dotnet/roslyn/pull/51855#discussion_r596330844) [](ancestors = 596330844)